[Agent builder] add support for workflow tools

pgayvallet opened the pull request:
## Summary

Fix https://github.com/elastic/search-team/issues/11316

Introduce a new type of tools, `workflow`, which allows agents to execute a specific workflow as a tool.

For now, the tool's schema is generated from the workflow's schema. We know that later we will want to let the user decide which parameters are exposed to the LLM versus which ones are statically defined by the user.


### How to test

1. enable all the things in your Kibana config

```yaml
workflowsManagement.enabled: true
workflowsExecutionEngine.enabled: true

uiSettings.overrides:
  agentBuilder:enabled: true
  workflows:ui:enabled: true
```

2. create a workflow

3. create a `workflow` tool and associate it with the created workflow

4. assign the tool to your favorite agent

5. ask the agent to use the tool

pgayvallet committed:
[agent builder] Workflow tools support - phase 1

pgayvallet committed:
add browser-side internal API

pgayvallet committed:
comment

pgayvallet committed:
fix imports

kibanamachine committed:
[CI] Auto-commit changed files from 'ts-node .buildkite/pipeline-resource-definitions/scripts/fix-location-collection.ts'

pgayvallet committed:
add workflow form data

pgayvallet committed:
starting to work on server-side implem

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

pgayvallet committed:
add base validation

joemcelroy committed:
[Agent Builder] add Workflows to ES3 (#235785)

## Summary

enables workflow in ES3. 

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

pgayvallet committed:
wip

pgayvallet committed:
first working version

Commit:
Merge branch 'main' into agent-builder-workflow-tools

jedrazb committed:
[Agent Builder] Add workflow tool type in UI (#235950)

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

pgayvallet committed:
fix UI types

pgayvallet committed:
retrieve available tool types from the server

pgayvallet committed:
ignore workflow settings in FTR test

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

pgayvallet committed:
replay autocommit

pgayvallet committed:
fix mock types

joemcelroy commented on src/platform/packages/shared/serverless/settings/search_project/index.ts:
```diff
@@ -25,4 +26,5 @@ export const SEARCH_PROJECT_SETTINGS = [
   OBSERVABILITY_AI_ASSISTANT_SIMULATED_FUNCTION_CALLING,
   AI_ANONYMIZATION_SETTINGS,
   AGENT_BUILDER_ENABLED_SETTING_ID,
+  WORKFLOWS_UI_SETTING_ID,
```
we will need to remove this for project settings because when workflow plugin isn't enabled, the uiSetting isn't registered and will complain on startup.

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

pgayvallet committed:
adapt for space support

pgayvallet commented:
/ci

pgayvallet committed:
remove workflow setting from search setting

pgayvallet committed:
evict unknown tools instead of throwing for now

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

pgayvallet committed:
Merge remote-tracking branch 'upstream/main' into agent-builder-workflow-tools

pgayvallet committed:
change workflow output details

pgayvallet committed:
remove dep

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/generate codeowners'

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343513)
* Commit: b2d92ebea6e29409baac29ddfc981efd9a0df2d0

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/62ea7a6e12f6bdb57527e12e4b00ee9bfb35ffc3) | [after](https://github.com/elastic/kibana/commit/b2d92ebea6e29409baac29ddfc981efd9a0df2d0) | diff |
| --- | --- | --- | --- |
| `onechat` | 589 | 597 | +8 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/62ea7a6e12f6bdb57527e12e4b00ee9bfb35ffc3) | [after](https://github.com/elastic/kibana/commit/b2d92ebea6e29409baac29ddfc981efd9a0df2d0) | diff |
| --- | --- | --- | --- |
| `onechat` | 305.4KB | 308.8KB | +3.5KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/62ea7a6e12f6bdb57527e12e4b00ee9bfb35ffc3) | [after](https://github.com/elastic/kibana/commit/b2d92ebea6e29409baac29ddfc981efd9a0df2d0) | diff |
| --- | --- | --- | --- |
| `onechat` | 15.6KB | 15.9KB | +289.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/62ea7a6e12f6bdb57527e12e4b00ee9bfb35ffc3) | [after](https://github.com/elastic/kibana/commit/b2d92ebea6e29409baac29ddfc981efd9a0df2d0) | diff |
| --- | --- | --- | --- |
| `@kbn/onechat-common` | 2 | 3 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/62ea7a6e12f6bdb57527e12e4b00ee9bfb35ffc3) | [after](https://github.com/elastic/kibana/commit/b2d92ebea6e29409baac29ddfc981efd9a0df2d0) | diff |
| --- | --- | --- | --- |
| `@kbn/onechat-common` | 2 | 3 | +1 |


</details>


### History
* :green_heart: [Build #343108](https://buildkite.com/elastic/kibana-pull-request/builds/343108) succeeded 12b87beac23cbdcb0c1fcb8b2c0d8fe55d6ab65a
* :green_heart: [Build #342936](https://buildkite.com/elastic/kibana-pull-request/builds/342936) succeeded 51adb0fa166cc0ccd2abf275b6c7cc8209e32b53
* :broken_heart: [Build #342563](https://buildkite.com/elastic/kibana-pull-request/builds/342563) failed f89bd743bd12dda41183efc343d6ab5912e3a6b9
* :broken_heart: [Build #342488](https://buildkite.com/elastic/kibana-pull-request/builds/342488) failed 4c9161ad1fe62c4bb9d98efdfcb2baa862d2e562
* :broken_heart: [Build #342392](https://buildkite.com/elastic/kibana-pull-request/builds/342392) failed d0d9d430ff000b770917f582f0a435680b31ad6f
* :broken_heart: [Build #342386](https://buildkite.com/elastic/kibana-pull-request/builds/342386) failed 0c401a6b03caacc3ada0f0e89e8db2034ff70dba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343513","number":343513,"commit":"b2d92ebea6e29409baac29ddfc981efd9a0df2d0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343108","number":343108,"commit":"12b87beac23cbdcb0c1fcb8b2c0d8fe55d6ab65a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342936","number":342936,"commit":"51adb0fa166cc0ccd2abf275b6c7cc8209e32b53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342563","number":342563,"commit":"f89bd743bd12dda41183efc343d6ab5912e3a6b9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342488","number":342488,"commit":"4c9161ad1fe62c4bb9d98efdfcb2baa862d2e562"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342392","number":342392,"commit":"d0d9d430ff000b770917f582f0a435680b31ad6f"}],"number":343513}
buildkite-pr-comment-->
